### ISSUE TITLE
Building issues with openssl on MacOSX HighSierra

### ISSUE THREAD

**Issue by woonhak** [libevent/libevent#667]
Hi,

I'm having some problem to build libevent-2.1.8-stable on mac with OpenSSL.
I have installed openssl using brew package manager and it is located in : `/usr/local/opt/openssl/`

I need to build this on multiple OS and arch, so I don't want to manually set cflags and ldflags in ENV.
I just want to use `pkg-config` to handle these paths generally.
so I ran configure with `PKG_CONFIG_PATH`,
and the OPENSSL_LIBS, OPENSSL_INCS were configured correctly.

Here's part of `config.status` 

```S["OPENSSL_FALSE"]="#"                                                        
S["OPENSSL_TRUE"]=""
S["OPENSSL_LIBS"]="-L/usr/local/Cellar/openssl/1.0.2o_2/lib -lssl -lcrypto   "
S["OPENSSL_INCS"]="-I/usr/local/Cellar/openssl/1.0.2o_2/include"
S["PKG_CONFIG"]="/usr/local/bin/pkg-config"
S["OPENSSL_LIBADD"]=""
```

However, I encountered building error because of missing headers, and it seems that it doens't  include OPENSSL_INCS in their compile command.
See the following errors.

```/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
depbase=`echo sample/le-proxy.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
        gcc -DHAVE_CONFIG_H -I.  -I./compat -I./include -I./include    -g -O2 -Wall -fno-strict-aliasing -Wno-deprecated-declarations -D_THREAD_SAFE  -W -Wfloat-equal -Wundef -Wpointer-arith -Wstrict-prototypes -Wmissing-prototypes -Wwrite-strings -Wredundant-decls -Wchar-subscripts -Wcomment -Wformat -Wwrite-strings -Wmissing-declarations -Wredundant-decls -Wnested-externs -Wbad-function-cast -Wswitch -Wno-unused-parameter -Wstrict-aliasing -Winit-self -Wmissing-field-initializers -Wdeclaration-after-statement -Waddress -Wno-unused-function -Qunused-arguments -MT sample/le-proxy.o -MD -MP -MF $depbase.Tpo -c -o sample/le-proxy.o sample/le-proxy.c &&\
        mv -f $depbase.Tpo $depbase.Po
sample/le-proxy.c:33:10: fatal error: 'openssl/ssl.h' file not found
#include <openssl/ssl.h>
         ^~~~~~~~~~~~~~~
1 error generated.
```

In the generated makefile, there's a `sample_le_proxy_INCLUDES` defined, can't find calling site. (Maybe I don't understand how this makefile using these, because I can't find call site for `sample_https_client_INCLUDES` too)

```sample_le_proxy_SOURCES = sample/le-proxy.c
sample_le_proxy_LDADD = libevent.la libevent_openssl.la ${OPENSSL_LIBS} ${OPENSSL_LIBADD} 
sample_le_proxy_INCLUDES = $(OPENSSL_INCS)
sample_https_client_SOURCES = \
    sample/https-client.c \
    sample/hostcheck.c \
    sample/openssl_hostname_validation.c
```

Is there any work-around to use pkg-config for libevent?

**Comment by ploxiln** [comment]
I think this was fixed in 0ec5edde503096a81831f36441a4115574ac45f0 (after 2.1.8 was released)

**Comment by woonhak** [comment]
@ploxiln Thanks a lot. It worked now.
